DEVINFO.md

```diff
@@ -0,0 +1,174 @@
+# How to Create and Review PRs
```

Comment:
I agree this should be in `CONTRIBUTING.md`. Looking at [github's communities docs](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file#supported-file-types), there's no mention at all of DEVINFO.md.

Reply:
We have a `CONTRIBUTORS.md`, which should probably be renamed `CONTRIBUTING.md`. There is some overlap between this file and that one, and the sections on logging, error handling, and testing could probably be merged into `CONTRIBUTING.md`. Both documents could link to each other for easy access.

Reply:
Wherever you put it is fine by me, just as long as it's easy to discover.

Reply:
I'll merge this into `docs/CONTRIBUTING.md` (and merge `CONTRIBUTORS.md` into that). That file gets some automatic visibility from Github (when opening an issue or a PR, the sidebar adds a link to those files, etc.)

Reply:
Do you think `CONTRIBUTING.md` be in the root directory like the other standard files (`README.md`, `LICENSE`, etc)? It might be easier for a new contributor to overlook it in `docs/`

Reply:
Moved from `docs/` to the top-level directory in 73f55526e